src/light_data/light_store.rs

```diff
@@ -0,0 +1,404 @@
+// This is a temporary shim which should be merged with scalar_store
+// Currently it only exists for reading store-dumps
+
+use crate::field::FWrap;
+#[cfg(not(target_arch = "wasm32"))]
+use proptest::prelude::*;
+#[cfg(not(target_arch = "wasm32"))]
+use proptest_derive::Arbitrary;
+
+use anyhow::anyhow;
+use std::collections::BTreeMap;
+
+use crate::light_data::Encodable;
+use crate::light_data::LightData;
+use crate::scalar_store::ScalarExpression;
+use crate::scalar_store::ScalarStore;
+use crate::store::ScalarPointer;
+use crate::store::ScalarPtr;
+use crate::sym::Sym;
+use crate::tag::ExprTag;
+
+use crate::field::LurkField;
+
+#[derive(Debug, PartialEq)]
+#[cfg_attr(not(target_arch = "wasm32"), derive(Arbitrary))]
+#[cfg_attr(not(target_arch = "wasm32"), proptest(no_bound))]
+/// LightStore contains a fragment of the ScalarStore, but using the `LightExpr` type
+pub struct LightStore<F: LurkField> {
+    /// An analogous to the ScalarStore's scalar_map, but with `LightExpr` instead of `ScalarExpression`
+    pub scalar_map: BTreeMap<ScalarPtr<F>, Option<LightExpr<F>>>,
+}
+
+impl<F: LurkField> Encodable for LightStore<F> {
+    fn ser(&self) -> LightData {
+        // TODO: this clone is loathsome
+        self.scalar_map
+            .clone()
+            .into_iter()
+            .collect::<Vec<(ScalarPtr<F>, Option<LightExpr<F>>)>>()
+            .ser()
+    }
+    fn de(ld: &LightData) -> anyhow::Result<Self> {
+        let pairs = Vec::<(ScalarPtr<F>, Option<LightExpr<F>>)>::de(ld)?;
+        Ok(LightStore {
+            scalar_map: pairs.into_iter().collect(),
+        })
+    }
+}
+
+impl<F: LurkField> LightStore<F> {
+    fn insert_scalar_string(
+        &self,
+        ptr0: ScalarPtr<F>,
+        store: &mut ScalarStore<F>,
+    ) -> anyhow::Result<String> {
+        let mut s = String::new();
+        let mut tail_ptrs = vec![];
+        let mut ptr = ptr0;
+        let strnil_ptr = ScalarPtr::from_parts(ExprTag::Str.as_field(), F::zero());
+
+        // TODO: this needs to bail on encountering an opaque pointer
```

Comment:
Is this 'needed' to keep the initial implementation simple, or is `LightStore` inherently incapable of handling opaque data?

Reply:
`LightStore` can handle opaque data no problem, the issue is if there's an opaque string tail there's not a good way to represent that in ScalarStore right now.